apiserver/params/instance_information.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package params
+
+import "github.com/juju/juju/constraints"
+
+// CloudInstanceTypesConstraints contains a slice of CloudInstanceTypesConstraint.
+type CloudInstanceTypesConstraints struct {
+	Constraints []CloudInstanceTypesConstraint `json:"constraints"`
+}
+
+// CloudInstanceTypesConstraint contains the cloud information and constraits
+// necessary to query for instance types on a given cloud.
+type CloudInstanceTypesConstraint struct {
+	// CloudTag is the tag of the cloud for which instances types
+	// should be returned.
+	CloudTag string `json:"cloud-tag"`
+
+	// CloudRegion is the name of the region for which instance
+	// types should be returned.
+	CloudRegion string `json:"region"`
+
+	// Constraints, if specified, contains the constraints to filter
+	// the instance types by. If Constraints is not specified, then
+	// no filtering by constraints will take place: all instance
+	// types supported by the region will be returned.
+	Constraints *constraints.Value `json:"constraints,omitempty"`
+}
+
+// ModelInstanceTypesConstraints contains a slice of InstanceTypesConstraint.
+type ModelInstanceTypesConstraints struct {
+	// Constraints, if specified, contains the constraints to filter
+	// the instance types by. If Constraints is not specified, then
+	// no filtering by constraints will take place: all instance
+	// types supported by the model will be returned.
+	Constraints []ModelInstanceTypesConstraint `json:"constraints"`
+}
+
+// ModelInstanceTypesConstraint contains a constraint applied when filtering instance types.
+type ModelInstanceTypesConstraint struct {
+	// Value, if specified, contains the constraints to filter
```

Comment:
Can you please change the name to Constraints, and tag to "constraints", to match the field in CloudInstanceTypesConstraints? That way one is a superset of the other.